### DISCUSSION TITLE
option to install a cask from a .pkg "for me only"

### DISCUSSION THREAD

**Discussion by itcarroll** [Homebrew/discussions#4860] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.17-9-g141ac15
ORIGIN: https://github.com/Homebrew/brew
HEAD: 141ac157fddd0dbd6c3331c8cc8751412a9acf05
Last commit: 8 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9f67002c8a8efa21db15e3194168b5f4810be3be
Core tap last commit: 6 hours ago
Core tap branch: master
Core tap JSON: 25 Oct 01:10 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--appdir=/Users/icarroll/Applications"]
HOMEBREW_EDITOR: emacs
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.6-x86_64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

The Quarto application is distributed via a .pkg, and the GUI asks if you would like to install "for all users" (i.e. needs sudo and installs to /Applications) or "only for me" (i.e. does not need sudo and installs to ~/Applications). I cannot find the incantation needed in the formula to mimic the "only for me" option.

Generally, I don't see anything in the casks documentation about installing into "~/Applications". There is the "--appdir" option, but it appears to be irrelevant for .pkg installers.

It's not hard to do this at the command line: `installer -pkg quarto-1.3.450-macos.pkg -target CurrentUserHomeDirectory`

Does it look like homebrew-cask would need a new feature to allow this, or is there an existing argument to the `pkg` stanza I can use?

### Relevant casks

quarto

**Comment by itcarroll** [comment]
Ah, well [here](https://github.com/Homebrew/brew/blob/3ebcbbb6b815486f51e2e0ae2505bf4a55c54e9a/Library/Homebrew/cask/artifact/pkg.rb#L52
) is what needs to be made into an option. That and possibly making [these](https://github.com/Homebrew/brew/blob/3ebcbbb6b815486f51e2e0ae2505bf4a55c54e9a/Library/Homebrew/cask/artifact/pkg.rb#L65-L66) `false`.

**Reply by SMillerDev** [reply]
It does depend on the installer actually supporting this.

**Comment by Kentzo** [comment]
Another angle is that some installers pretend to install packages globally while in reality they tie installation to the user that runs it. [IBKR Desktop](https://github.com/Homebrew/homebrew-cask/edit/master/Casks/i/ibkr.rb) is an example. Possibly this extends to all packages generated by [`install4j`](https://www.ej-technologies.com/install4j).

Homebrew Casks should acknowledge this by allowing users to set `--target-user`.
